restaurant/pkg/api/order.go

```diff
@@ -0,0 +1,123 @@
+package api
+
+import (
+	"encoding/json"
+	"github.com/miladouski/golang-training-restaurant/restaurant/pkg/data"
+	"log"
+	"net/http"
+	"strconv"
+
+	"github.com/gorilla/mux"
+)
+
+type orderAPI struct {
+	data *data.OrderData
+}
+
+func ServeOrderResource(r *mux.Router, data data.OrderData) {
+	api := &orderAPI{data: &data}
+	r.HandleFunc("/orders", api.getAllOrders).Methods("GET")
+	r.HandleFunc("/orders", api.createOrder).Methods("POST")
+	r.HandleFunc("/orders/{id}", api.FindOrder).Methods("GET")
+	r.HandleFunc("/orders/{id}", api.updateOrder).Methods("PUT")
+	r.HandleFunc("/orders/{id}", api.deleteOrder).Methods("DELETE")
+}
+
+func (o orderAPI) getAllOrders(writer http.ResponseWriter, request *http.Request) {
+	orders, err := o.data.ReadAll()
+	if err != nil {
+		_, err := writer.Write([]byte("got an error when tried to get orders"))
+		if err != nil {
+			log.Println(err)
```

Comment:
You just printing an error, and not exit from function, so in step: encoding go model to json you will get an error again.
So please add writer.WriteHeader(http.StatusInternalServerError) and exit of function after printing log message.